### ISSUE TITLE
Unstable Aerostack2 Project

### ISSUE THREAD

**Issue by zZ-kunja-Zz** [aerostack2/project_crazyflie#13]
I am reaching out regarding some issues we are encountering with the integration of AeroStack2 and Crazyflie drones in our project. We are using a setup involving Ubuntu 22, AeroStack2, Motive 3.02, and 16 Optitrack prime 22 cameras for motion capture. The ROS2 humble project includes the standard AeroStack2 package, the motion capture package, and a simple Crazyflie example project.

We use the [mocap.launch.py](http://mocap.launch.py/), which functions correctly after we run the launch command on the simple crazyflie project. To fly the drone, we use the sample mission.py. First the parts of the project we know to work. The controller functions correctly, the drones receive their pose correctly, and the radio connects correctly. When starting up the project, we run into three situations:

1. Where the drone will fly correctly every day but must be turned off between trials, and the launch file has to be closed and reopened.
2. The drone flies, but there is a varying delay in receiving pose data.
3. The drone does not attempt to take off. These three situations will stay until the project is rebuilt and everything is turned off.
Additionally, we sometimes get takeoff behavior not available warnings but this does not seem to affect performance.

We would greatly appreciate any guidance on troubleshooting these issues or any known solutions that could help resolve them.

**Comment by zZ-kunja-Zz** [comment]
I believe our issue stems from not understanding how the launch files mocap.launch.py, tmuxinator start and ./launch_as2.bash work together and how they should be nested or separated into different terminals and in which order to use them in.

**Comment by pariaspe** [comment]
Hi, sorry for the late response,

>     1. Where the drone will fly correctly every day but must be turned off between trials, and the launch file has to be closed and reopened.

After, running `launch_as2.bash` you should be able to run mission scripts multiple times unless you do an emergency kill switch, where full aerostack2 and crazyflie have to be reboot.

>     2. The drone flies, but there is a varying delay in receiving pose data.

Might it be your system causing the delay? How big is the delay and how are you detecting it? It seems to be an issue related with your system or ROS2, not with aerostack2. 
 
>     3. The drone does not attempt to take off. These three situations will stay until the project is rebuilt and everything is turned off.
>        Additionally, we sometimes get takeoff behavior not available warnings but this does not seem to affect performance.

Can you provide some more details? It is hard to guess why the drone might not taking off with the brief description. Takeoff (or other behaviors) warnings are due to ROS2 discovery protocol which fails sometimes, you can skip those warnings most of the times

**Comment by zZ-kunja-Zz** [comment]
Thank you for your reply,

I think it's important to note that these trials were run without the tmuxinator command. When running with the tmuxinator command there was no way for us to fly the drone, despite the mocap/rigid_bodies topic outputting pose correctly and the drone localization data having the pose. The drone was drifting off into walls. as if the controller did not have the pose. Perhaps this is an issue with the setting of the external odometry variable in the platform_config_mocap.yaml which we set to /mocap/rigid_bodies along with the cr0_state_estimator.yaml where we set the mocap_topic to /mocap/rigid_bodies. Were we wrong to attempt trials without the tmuxinator file, it was confusing to me how the drone seemed to work better without it. It would be helpful if there was an example project on Github for me to look through to compare.
